File: Task 1/Java/app/src/main/java/dev/ogabek/java/database/NoteDao.java
package dev.ogabek.java.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import dev.ogabek.java.model.Note;

@Dao
public interface NoteDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long saveData(Note note);

    @Query("SELECT * FROM note_table")
    List<Note> getAllNotes();

    @Query("DELETE FROM note_table")
    void clearData();

}

File: Task 1/Kotlin/app/src/main/java/dev/ogabek/kotlin/database/NoteDao.kt
package dev.ogabek.kotlin.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import dev.ogabek.kotlin.model.Note

@Dao
interface NoteDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveNote(note: Note)

    @Query("SELECT * FROM note_table")
    fun getAllNotes(): List<Note>

    @Query("DELETE FROM note_table")
    fun deleteData()

}